html cards for people section

CleverFool77 opened the pull request:
Fixes #4922 (<=== Add issue number here)

## Earlier Description
~~People page will be redesigned soon. And one the requirements was that all the users should be listed down in grid card format instead rows in table. I started off with the basic structure of cards for users. And soon the further improvements will be done.~~

![Screenshot from 2019-04-03 19-34-10](https://user-images.githubusercontent.com/26685258/55487413-ac9cbd80-564b-11e9-919e-f8fe1c440fe3.png)

![h](https://user-images.githubusercontent.com/26685258/55487324-837c2d00-564b-11e9-9b9e-f3ab7d195f0a.gif)

## Update
This is the new redesign of people's page.
I made it two column and map on right. The map remain stick on scrolling.
I showed the users on left
On md and lg scrrens - 
![h](https://user-images.githubusercontent.com/26685258/55497323-c398db00-565e-11e9-9dc6-7d2a98f2e194.gif)


On small screens - 
![hh](https://user-images.githubusercontent.com/26685258/55497354-d27f8d80-565e-11e9-8909-2b40c4d8a8be.gif)


* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below


Thanks!

CleverFool77 commented:
@gauravano i tried the following to pass the tests
since now there is no static value in header to check for existence, i tried asserting usernames of user but i wasn't able to pass the tests. I fetched the users and stored in a variables user_data = users(). I asserted as user_data.username().to_s inside a loop. but the issue was the user in this way appeared as `<spammer>`
Then i tried checking for the existence of the row in which all cards are going. I gave that div an id but I wasn't able to find how to check existence of a div with specific id in ruby test.

How can I proceed to pass the tests.

jywarren commented:
Hi, going to try to help with the errors too, but also wanted to say - this is very exciting! Thanks!

I wonder if we can continue refining the card display even more; for example:

![image](https://user-images.githubusercontent.com/24359/54855191-c7764600-4ccb-11e9-8c10-13b36ca53dbf.png)![](https://publiclab.org/system/images/photos/000/029/338/original/Screen_Shot_2019-02-14_at_6.52.03_PM.png)

Could we:

1. add a bit more padding/margins inside so that it matches the mockup a bit more?
2. make the username text black and a larger size, and Junction font?
3. make the rest of the text smaller and grey, with a bit more line spacing?

This is very cool! Its going to be awesome!

jywarren commented:
Ok, do you think you could push up the tests you've written so far? Here are a few examples of some CSS stuff you can `assert_select`, but it's not quite as flexible as jQuery in fetching specific CSS:

https://github.com/publiclab/plots2/blob/master/test/functional/notes_controller_test.rb#L56-L57

https://github.com/publiclab/plots2/blob/master/test/functional/notes_controller_test.rb#L157-L158

I hope this helps!

CleverFool77 commented:
Hi @jywarren. Thanks I for the help. I'll follow the methods in links. Regarding the cards,I wa thinking of making basic layout first and then refining them but now that you've mentioned I'll make them look more like the ui in the design given.
Thanks for the review :)

CleverFool77 commented:
Hi @jywarren and @gauravano 
I updated this PR.
I redesigned the basic structure of the people page
I was confused regarding the card ui that which one to choose, considering there were two designs.
![s](https://user-images.githubusercontent.com/26685258/55494602-be389200-5658-11e9-85fe-a9c145fd474e.png)
and 
![ss](https://user-images.githubusercontent.com/26685258/55494632-c7296380-5658-11e9-84a2-b27bce68e609.png)

For now, I went for the ui of card in the above comments.
I removed the test for I18n for `'users.list.members_last_activity` , as the data to be tested has been removed.
Regarding the functional test, I'll write them to check card rendering.

gauravano commented:
Lol! both of them are awesome! I can't choose one :see_no_evil:. Let's ask @publiclab/community-reps

gauravano commented:
Although, have you implemented 2nd one on the website, yet?

CleverFool77 commented:
Hi @gauravano 
I implemented second one as warren comment in above comments to make the ui similar to it. But Now if I'll go for grid system of two column with map in right side. I'm not sure which UI is chosen.

CleverFool77 commented:
Hi again,
The update in the PR description is latest changes I've pushed.
I didn't remove the Earlier description as I needed two choices for comparison.

CleverFool77 commented:
Hi @jywarren @gauravano I've added the test for browsing people's page and card rendering.

jywarren commented:
This is looking amazing! However, I maybe want to caution that it could be better to work in smaller pieces -- you've done some great work here, but while we could have merged the card UI now, we should take a few other steps on the other parts, before merging. So, since it's all in one PR, we will need to hold a little bit. Doing it in smaller parts is also great for sharing the work among people! But that's OK this time, just a note for next time! 🙌 👍 🎉 

Note a few things here. The UI you've done is super. Now take a look at this screenshot of the existing:

![image](https://user-images.githubusercontent.com/24359/55657819-7ab16580-57c9-11e9-9ec8-c20a9fd61364.png)

What we should think about is:

1. the table sorting. Can we preserve it as was done on this page? https://github.com/publiclab/plots2/pull/5239
2. there is a "ban" button shown to admins/moderators. Could this be put after 'joined ____ ago' for now, until we do a follow-up PR (modularity in mind 😄 ) for the `...` menu?

What do you think? 

Thank you @CleverFool77 !!! This is very cool!

CleverFool77 commented:
Hi @jywarren I started off with basic card-layout in this PR but as I procedded to make the card look like the provided ui by team. It itself turned into this. 
Regarding sorting and other, It was left as next part. Or should the sorting part be done here only ?
regarding the ban button, it's there I'll provide the gif after logging as admin. As right now In the gif I uploaded, I logged in as new contributor.
Tbh earlier my aim for this PR was a baisc card layout but as I proceeded, it turned into something major.

Thanks !!!!!

CleverFool77 commented:
I did some fixes to size of ban button
And here is the screenshot
**Desktop**
![Screenshot from 2019-04-06 11-34-59](https://user-images.githubusercontent.com/26685258/55665598-3cb35080-5860-11e9-9467-645754237cd5.png)

**Mobile interface**
![Screenshot from 2019-04-06 11-35-23](https://user-images.githubusercontent.com/26685258/55665600-4a68d600-5860-11e9-99fc-7c32652a438a.png)

CleverFool77 commented:
Hi @jywarren @gauravano ,
Here is the latest changes that I've comitted.

**Admin**
![dd](https://user-images.githubusercontent.com/26685258/55666232-28c01c80-5869-11e9-8a74-767dfe38be39.gif)

**User**
![hh](https://user-images.githubusercontent.com/26685258/55666321-ca943900-586a-11e9-9c9f-dc159edf2302.gif)

jywarren commented:
This looks really good!!! I am wondering, if it's possible to wait until after we merge the giant #3937 Bootstrap 4 upgrade, and then be sure this works on Bootstrap 4 styles too? What do you think?

We're trying to all collaborate on identifying the final issues with the upgrade in this issue: https://github.com/publiclab/plots2/issues/5182

SidharthBansal commented:
I think we should merge #3937 first.

jywarren commented:
OK! https://github.com/publiclab/plots2/pull/3937/ was merged!!!!! @CleverFool77 would you want to try updating over that change and ensuring Bootstrap 4 compatibility? Thank you!!!

CleverFool77 commented:
Hi @jywarren 
I'll update it soon.

SidharthBansal commented:
No rush!!! Best of luck for your exams

SidharthBansal commented:
There are some conflicts in your pr. It will be great if you can rebase your pr whenever you are free.
Thanks for the work done here at PL

CleverFool77 commented:
I seem to have mess up my this branch very badly. 
I'm closing this PR and will be opening a new one.
Thanks !!

CleverFool77 commented:
Hi,
I resolved few stuffs here and did changes as per bootstrap4 upgrade. 
I need some review regarding the changes.
@jywarren @SidharthBansal @gauravano 
Thanks !!!

jywarren commented:
This is looking great. Can you share a screenshot? Thanks!!

CleverFool77 commented:
Hi @jywarren 
I need some suggestions regarding following points -
* The side margins. It should be decreased ?
* The cards look elongated vertically. 
* Make map sticky ? 

![people](https://user-images.githubusercontent.com/26685258/57465791-fad15c00-729c-11e9-9786-cb883cf7d67e.gif)

jywarren commented:
Oh cool! Let's leave the map as is for now, as we may want to reconfigure a
little bit later with more guidance from the style guide. I think the card
layout could also try to get closer to the style guide designs although the
style guide is not actually yet written in bootstrap... so that'll be an
ongoing process.

Finally, the margins look ok to me. Thanks!!

On Thu, May 9, 2019, 11:31 AM Lekhika Dugtal <notifications@github.com>
wrote:

> Hi @jywarren <https://github.com/jywarren>
> I need some suggestions regarding following points -
>
>    - The side margins. It should be decreased ?
>    - The cards look elongated vertically.
>    - Make map sticky ?
>
> [image: people]
> <https://user-images.githubusercontent.com/26685258/57465791-fad15c00-729c-11e9-9786-cb883cf7d67e.gif>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/5169#issuecomment-490952883>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAF6J7F3MP562UQH4QRITTPUQ7SXANCNFSM4G7NIRIA>
> .
>

CleverFool77 commented:
Hi @jywarren 
I went through style guide link.
The Design seems bit different. It's not two equal column anymore. And few other changes.
And regarding the card too. There were two card design. On the basis of conversation happened above in this PR. I started working on vertical one. So Should I go for horizontal one ?
Should I do it like as it's mentioned in style guide ? or should I just wait for sometime as there would be many more suggestions and changes in style guide, I guess.

jywarren commented:
Sure, we can work towards the style guide incrementally, and also see as it
gets finalized. In general, i'm thinking that just a bit less padding
inside the cards would be helpful. And we can work on the column spacing
later as well. Thanks!

On Thu, May 9, 2019 at 1:16 PM Lekhika Dugtal <notifications@github.com>
wrote:

> Hi @jywarren <https://github.com/jywarren>
> I went through style guide link.
> The Design seems bit different. It's not two equal column anymore. And few
> other changes.
> And regarding the card too. There were two card design. On the basis of
> conversation happened above in this PR. I started working on vertical one.
> So Should I go for horizontal one ?
> Should I do it like as it's mentioned in style guide ? or should I just
> wait for sometime as there would be many more suggestions and changes in
> style guide, I guess.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/5169#issuecomment-490990026>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAF6J7QDKBGIYY3CP3MM2DPURL3HANCNFSM4G7NIRIA>
> .
>

CleverFool77 commented:
I decreased the padding for card-info body by `0.5em` on left and right. For bottom and top, I decreased it by `0.7em` and `1em`.

jywarren commented:
This looks great! I'll merge this once it passes. Thanks a lot, @CleverFool77 !!!
